frontend/test/__support__/server-mocks/action.ts

```diff
@@ -12,17 +12,27 @@ export function setupActionEndpoints(action: WritebackAction) {
 }
 
 export function setupActionsEndpoints(
-  modelId: CardId,
   actions: WritebackAction[],
+  modelId?: CardId,
 ) {
-  fetchMock.get(
-    {
-      url: "path:/api/action",
-      query: { "model-id": modelId },
-      overwriteRoutes: false,
-    },
-    actions,
-  );
+  if (modelId) {
+    fetchMock.get(
+      {
+        url: "path:/api/action",
+        query: { "model-id": modelId },
+        overwriteRoutes: false,
+      },
+      actions,
+    );
+  } else {
+    fetchMock.get(
+      {
+        url: "path:/api/action",
+        overwriteRoutes: false,
+      },
+      actions,
+    );
+  }
```

Comment:
@kulyk what do you think about this approach to making this mock work for this endpoint with or without a query param?

Part of me hates adding conditionals here, because the more logic in the mock, the more we're just testing the mock. But I think this is probably OK for the sake of fitting into existing patterns.

Reply:
@iethree what do you think about something like that?

```js
function setupActionsEndpoints(actions) {
  fetchMock.get("/api/action", actions);

  const byModelId = _.groupBy(actions, "model_id");
  Object.keys(byModelId).forEach(modelId => {
    const modelActions = byModelId[modelId];
    fetchMock(`/api/action?model-id=${modelId}`, modelActions);
  });
  
  actions.forEach(action => setupActionEndpoints(action));
}
```

I think it's fine to just setup all the endpoints out of the box. A single `setupDatabasesEndpoints` can add much more mocks

Reply:
😬  this is the sort of logic that makes me nervous that it could lead to test fragility down the road, or end up with us mostly testing the logic of the mock. IMHO, a good mock is simple and obvious, and should generally just give you back whatever data you put into it.

Reply:
Yeah, that's thin ice 😢 As an alternative, should we maybe make them two different functions? Feels off that a single function can set up two different endpoints conditionally. What I mean:

```js
function setupActionsEndpoints(actions) {
  fetchMock.get("/api/action", actions);
  actions.forEach(action => setupActionEndpoints(action));
}

function setupModelActionsEndpoints(modelId, actions) {
  fetchMock.get(`/api/action?model-id=${modelId}`, actions);
  actions.forEach(action => setupActionEndpoints(action));
}
```